src/EditorFeatures/Core/Implementation/CodeFixes/CodeFixService.cs

```diff
@@ -171,8 +174,8 @@ public async Task<ImmutableArray<CodeFixCollection>> GetFixesAsync(Document docu
 
                 cancellationToken.ThrowIfCancellationRequested();
 
-                aggregatedDiagnostics ??= new SortedDictionary<TextSpan, SortedList<string, DiagnosticData>>();
-                aggregatedDiagnostics.GetOrAdd(diagnostic.GetTextSpan(), _ => new SortedList<DiagnosticId, DiagnosticData>()).Add(diagnostic.Id, diagnostic);
+                aggregatedDiagnostics ??= new SortedDictionary<TextSpan, SortedSet<DiagnosticData>>();
+                aggregatedDiagnostics.GetOrAdd(diagnostic.GetTextSpan(), _ => new SortedSet<DiagnosticData>(s_diagnosticDataComparerById)).Add(diagnostic);
```

Comment:
❗️ This will silently discard diagnostics that occur and the same location with the same ID.

Reply:
I just found the same and wrote a unit test for it :-). Will push a commit soon.

Reply:
Expanded the added unit test, so that the newly added assert fails with the prior fix, and succeeds with the modified fix.